x/bridge/observer/osmosis/osmosis_test.go

```diff
@@ -0,0 +1,323 @@
+	require.NoError(t, err)
+	expResp1 := &authtypes.QueryAccountResponse{
+		Account: &codectypes.Any{
+			Value: m,
+		},
+	}
+	ots.ts.AccServer.
+		EXPECT().
+		Account(gomock.Any(), expReq1).
+		Times(1).
+		Return(expResp1, nil)
+
+	keyring := keyring.NewInMemory(app.GetEncodingConfig().Marshaler)
+	_, err = keyring.NewAccount(
+		osmosis.ModuleNameClient,
+		Mnemonic1,
+		"",
+		types.FullFundraiserPath,
+		hd.Secp256k1,
+	)
+	require.NoError(t, err)
+
+	in := observer.Transfer{
+		SrcChain: observer.ChainIdBitcoin,
+		DstChain: observer.ChainIdOsmosis,
+		Id:       "deadbeef",
+		Height:   42,
+		Sender:   Addr1.String(),
+		To:       BtcAddr,
+		Asset:    "btc",
+		Amount:   math.NewUint(10),
+	}
+	msg := bridgetypes.NewMsgInboundTransfer(
+		in.Id,
+		in.Sender,
+		in.To,
+		bridgetypes.AssetID{
+			SourceChain: string(in.SrcChain),
+			Denom:       in.Asset,
+		},
+		math.Int(in.Amount),
+	)
+	fees := sdktypes.NewCoins(sdktypes.NewCoin(osmosis.OsmoFeeDenom, osmosis.OsmoFeeAmount))
+	expBytes := buildAndSignTx(
+		t,
+		keyring,
+		expectedAcc.AccountNumber,
+		expectedAcc.Sequence,
+		msg,
+		fees,
+		osmosis.OsmoGasLimit,
+	)
+	expReq2 := &tx.BroadcastTxRequest{
+		TxBytes: expBytes,
+		Mode:    tx.BroadcastMode_BROADCAST_MODE_SYNC,
+	}
+	expResp2 := &tx.BroadcastTxResponse{
+		TxResponse: &types.TxResponse{
+			Height: 50,
+			TxHash: "deadbeef",
+		},
+	}
+	ots.ts.TxServer.
+		EXPECT().
+		BroadcastTx(gomock.Any(), gomock.Eq(expReq2)).
+		Times(1).
+		DoAndReturn(func(
+			ctx context.Context,
+			req *tx.BroadcastTxRequest,
+		) (*tx.BroadcastTxResponse, error) {
+			fmt.Println("BroadcastTx")
+			return expResp2, nil
+		})
+	ots.Start(t, ctx)
+
+	err = ots.o.SignalInboundTransfer(ctx, in)
+	require.NoError(t, err)
+
+	ots.Stop(t, ctx)
+}
+
+// ListenOutboundTransfer verifies Osmosis properly collects transfers
+// from the chain and sends it into the outbound channel
+func TestListenOutboundTransfer(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	ots := NewOsmosisTestSuite(t, ctx)
+
+	height, err := ots.o.Height()
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), height)
+	ots.Start(t, ctx)
+
+	// We expect to receive 3 Txs with `EventOutboundTransferType` events in this test
+	// Only 2 of the Txs are successful, so we should receive only 2 event through the channel
+	eventsOut := ots.o.ListenOutboundTransfer()
+	transfers := [2]observer.Transfer{}
+	for i := 0; i < len(transfers); i++ {
+		require.Eventually(t, func() bool {
+			transfers[i] = <-eventsOut
+			return true
+		}, time.Second, 100*time.Millisecond, "Timeout reading events from observer")
+	}
+
+	expTransfer0 := observer.Transfer{
+		SrcChain: observer.ChainIdOsmosis,
+		DstChain: observer.ChainIdBitcoin,
+		Id:       "E765E65A3A513CCC3E2CE25BB6B47DBD7CA09AC6C7C380B84D96B88B3B0B8A70",
+		Height:   5984109,
+		Sender:   Addr1.String(),
+		To:       BtcAddr,
+		Asset:    "btc",
```

Comment:
In prod, will this be the tokenfactory btc denom or will this still be btc?

Reply:
This should be a bridge denom. Fixed variables in the next PR https://github.com/osmosis-labs/osmosis/pull/7882